dockerfiles/sv-base-mini/Dockerfile

```diff
@@ -1,77 +1,102 @@
-    samtools --help && \
-    bcftools --help
+        ca-certificates autoconf automake bzip2 g++ make wget pkgconf python2 \
+        libssl-dev libbz2-dev libcurl4-openssl-dev liblzma-dev libncurses-dev zlib1g-dev libdeflate-dev
 
-############### stage 1: other commonly used bioinformatics utilities 
-FROM ubuntu:18.04
+# install samtools
+ARG HTSLIB_VERSION
+RUN wget -q https://github.com/samtools/samtools/releases/download/${HTSLIB_VERSION}/samtools-${HTSLIB_VERSION}.tar.bz2 && \
+    tar xjf samtools-${HTSLIB_VERSION}.tar.bz2 && \
+    cd samtools-${HTSLIB_VERSION} && \
+    ./configure --quiet --enable-libcurl --prefix=/opt/samtools && \
+    make -s -j $(nproc --all) all all-htslib && \
+    make -s install install-htslib
+# install bcftools
+RUN wget -q https://github.com/samtools/bcftools/releases/download/${HTSLIB_VERSION}/bcftools-${HTSLIB_VERSION}.tar.bz2 && \
+    tar xjf bcftools-${HTSLIB_VERSION}.tar.bz2 && \
+    cd bcftools-${HTSLIB_VERSION} && \
+    ./configure --quiet --prefix=/opt/bcftools && \
+    make -s -j $(nproc --all) && \
+    make -s install
+# install vcftools
+ARG VCFTOOLS_VERSION
+RUN wget -q https://github.com/vcftools/vcftools/releases/download/v$VCFTOOLS_VERSION/vcftools-${VCFTOOLS_VERSION}.tar.gz && \
+    tar zxf vcftools-${VCFTOOLS_VERSION}.tar.gz && \
+    cd vcftools-${VCFTOOLS_VERSION} && \
+    export PERL5LIB=$(pwd)/src/perl/ && \
+    ./configure --quiet --prefix=/opt/vcftools && \
+    make -s -j $(nproc --all) && \
+    make -s install
+
+# install bedtools
+#   during install, bedtools needs python2 to be callable as "python"
+RUN ln -s /usr/bin/python2 /usr/bin/python
+ARG BEDTOOLS_VERSION
+RUN wget -q https://github.com/arq5x/bedtools2/releases/download/v$BEDTOOLS_VERSION/bedtools.static.binary && \
+    mkdir -p /opt/bedtools/bin && \
+    mv bedtools.static.binary /opt/bedtools/bin/bedtools && \
+    chmod a+x /opt/bedtools/bin/bedtools
+
+
+############### stage 1: copy tools and install needed non-dev libraries
+FROM ubuntu:$UBUNTU_RELEASE
 
 ENV HOME=/root
 
 # Define default command.
 CMD ["bash"]
 
-# copy from previous stage the binaries from samtools build
-COPY --from=0 /usr/local/bin/* /usr/local/bin/
+# copy utility file for cleaning up $APT_REQUIRED_PACKAGES
+RUN mkdir -p /opt/bin
+COPY dockerfiles/sv-base-mini/fix_spaces.sh /opt/bin/
+# copy the results of the build stage, and update path
+COPY --from=build /opt /opt
+ENV PATH=/opt/samtools/bin:/opt/bcftools/bin:/opt/vcftools/bin:/opt/bedtools/bin:/opt/bin:$PATH
+# note: need to specify path to perl libraries or tools from vcftools won't work
+RUN LIB_LOC=$(find /opt/vcftools/share/perl -mindepth 1 -type d) && \
+    ln -s $LIB_LOC /opt/vcftools/share/perl_lib
+ENV PERL5LIB=/opt/vcftools/share/perl_lib
 
-#### Basic utilities
+# establish file storing required apt packages (so subsequent stages know which packages can be removed and which can't)
+ARG APT_REQUIRED_PACKAGES
+ENV APT_REQUIRED_PACKAGES=$APT_REQUIRED_PACKAGES
```

Comment:
A bit confusing - do you really need the ENV here? Could at least give them different names